web/source/kmwembedded.ts

```diff
@@ -374,18 +374,25 @@ namespace com.keyman.text {
       /* Clear any pending (non-popup) key */
       osk.vkbd.keyPending = null;
 
-      // Changes for Build 353 to resolve KMEI popup key issues      
+      // Changes for Build 353 to resolve KMEI popup key issues
       keyName=keyName.replace('popup-',''); //remove popup prefix if present (unlikely)
 
       // Can't just split on '-' because some layers like ctrl-shift contain it.
       let separatorIndex = keyName.lastIndexOf('-');
-      var layer = core.keyboardProcessor.layerId;
+      var displayLayer = core.keyboardProcessor.layerId;
       if (separatorIndex > 0) {
-        layer = keyName.substring(0, separatorIndex);
+        displayLayer = keyName.substring(0, separatorIndex);
         keyName = keyName.substring(separatorIndex+1);
       }
-      if(layer == 'undefined') {
-        layer=core.keyboardProcessor.layerId;
+      if(displayLayer == 'undefined') {
+        displayLayer=core.keyboardProcessor.layerId;
+      }
+
+      // Determine the "functional" layer
+      let layer = 'default';
```

Comment:
```suggestion
      let layer = displayLayer;
```

Reply:
Relevant code for the non-embedded representation:

https://github.com/keymanapp/keyman/blob/7db9477f85c8dee3585477e2ecf4e68c5028a4fd/web/source/osk/browser/subkeyPopup.ts#L93-L102

`vkbd.layerId` matches `displayLayer` in purpose.